src/analysis/analyzer.ts

```diff
@@ -69,11 +71,26 @@ export class Analyzer extends AnalyzerGen {
 	}
 
 	private sendMessage<T>(req: Request<T>) {
-		let json = JSON.stringify(req);
-		if (config.verbose)
-			console.log(`==> ${json}`);
+		let json = JSON.stringify(req) + "\n";
+		this.logTraffic(`==> ${json}`);
 		this.analyzerProcess.stdin.write(json);
-		this.analyzerProcess.stdin.write("\n");
+	}
+
+	private logTraffic(message: String): void {
+		const max: number = 2000;
+
+		if (config.analyzerLogFile) {
+			if (!this.logStream)
+				this.logStream = fs.createWriteStream(config.analyzerLogFile);
+			if (message.length > max)
+				this.logStream.write(message.substring(0, max) + "...\n");
+			else
+				this.logStream.write(message);
+		} else if (!config.analyzerLogFile && this.logStream) {
+			// Turn off logging.
+			this.logStream.close();
```

Comment:
I don't know if it's important, but this `logStream` will only be closed if the setting is turned off and another request sent. Could this result in a locked file if you reload window / change workspace folder? Would it be better closing in the `close` method if open too?

(though really I wonder if this class should've been disposable and adding to `vscode.subscriptions` instead of a close method!)